Fix: MXP encoding corruption with incomplete numeric tags

mpconley opened the pull request:
#### Brief overview of PR changes/additions

Fixed encoding corruption issue in MXP (MUD eXtension Protocol) parser where incomplete numeric tags like `<33` caused question mark characters to appear in subsequent text. The fix simplifies error recovery to preserve original byte values instead of attempting encoding conversion.

#### Motivation for adding to Mudlet

Users reported seeing garbled text with question mark characters appearing after encountering incomplete MXP tags in MUD output. This affected readability and user experience, particularly with non-ASCII characters. The issue was corrupting character encoding during MXP error recovery.

#### Other info (issues closed, discussion etc)

- Reported by @Sanuki in Discord: https://discord.com/channels/283581582550237184/427919962561052673/1445185602847772815
- Added comprehensive test case to prevent regression
- All existing MXP tests continue to pass (35/35)
- Preserves compatibility with valid numeric MXP tags per specification

mpconley committed:
initial

Sanaki commented:
Very close, but not quite. It still eats a newline.
<img width="744" height="75" alt="image" src="https://github.com/user-attachments/assets/fd80e3c4-897b-410c-bb8d-5c77509d03ec" />

EDIT: And when you have a single-line prompt, it still produces encoding errors. And the lines with those errors are not visible until you get -another- line from the MUD.
<img width="488" height="90" alt="image" src="https://github.com/user-attachments/assets/f87e6e10-7a47-4676-89de-84cc5a5c0a15" />

mpconley committed:
Merge branch 'development' into fix/mxp-encoding-corruption

mpconley committed:
Fix: Comprehensive MXP encoding corruption prevention

Expanded the fix to address encoding corruption in multiple scenarios:
- Normal tag processing path for unhandled tags
- MXP watchdog timeout recovery mechanism
- Additional test coverage for malformed tag scenarios

This addresses the remaining issues reported by @Sanaki including:
- Newline handling problems
- Single-line prompt encoding errors
- Display timing issues

All MXP processing paths now use Latin1 encoding preservation
to avoid character corruption artifacts.

Tests: 36/36 passing (added testMalformedTagsPreserveEncoding)

mpconley committed:
Add bounds checking to TEncodingHelper and improve MXP encoding tests

- Add bounds checking to TEncodingHelper::decodeWithLookupTable() to prevent
  array access violations with Unicode replacement character fallback
- Update MXP encoding tests to use proper UTF-8 conversion instead of toLatin1()
- Make Unicode preservation test stricter to enforce full 'café' preservation
- Ensure consistent UTF-8 handling across both encoding corruption test methods

Sanaki commented:
Still no dice. Different encoding garbage, but still encoding garbage, and still eats a linebreak on the double prompt. I actually think it took more sent lines before it would show me anything this time (four).
<img width="712" height="90" alt="image" src="https://github.com/user-attachments/assets/15f14be0-f4e8-4537-abe4-89b36125523c" />

<img width="632" height="135" alt="image" src="https://github.com/user-attachments/assets/c8dbef4a-4fbb-4e15-8f58-e7d8531b1dc7" />

mpconley commented:
@Sanaki I believe I am trying out the same as you., please validate.

* Are you using the builds listed above to get your results?
* Are you certain there is not a GUI element sending you this garbage comment? I see those Comm.Channel.Start messages flooding into your main console.

<img width="834" height="439" alt="Screenshot 2025-12-02 at 8 47 49 PM" src="https://github.com/user-attachments/assets/5eb6783f-679d-4260-9b9d-72a34224f2b7" />

Sanaki commented:
I am indeed using the appimage from the above links, and that Comm.Channel echo is just a basic echo I have from a script I never finished.  To verify though, I just hit the emergency stop and tried sending `'<3` and `say <3` again (just in case using the apostrophe shortcut was somehow involved). In both cases I got garbage back.
<img width="768" height="150" alt="image" src="https://github.com/user-attachments/assets/8ecf56b8-e46b-46fe-a8cf-6959067f246e" />

Sanaki commented:
Ahhh, hold up, found the issue. This is occurring when I have MXP processing forced on in my preferences, but CONFIG MXP off in the game.
